tests/tensor/test_basic.py

```diff
@@ -4608,3 +4609,23 @@ def core_np(x, y):
         vectorize_pt(x_test, y_test),
         vectorize_np(x_test, y_test),
     )
+
+
+@pytest.mark.parametrize(
+    "ift, iff",
+    [(None, None), (7, 10), (7, None)],
+    ids=["both none", "both valid", "one none"],
+)
+def test_where_for_only_condition(ift, iff):
```

Comment:
Nitpick: this is a case where I don't think parametrize gains you anything you are not reusing almost any of the lines across parametrizations.

I would do a single sequential test that goes over the 3 cases.

My rule of thumb is I use parametrize when I would want to write a loop inside the test function to test multiple cases. If I would rather check things with different looking code one after the other, I wouldn't.

Reply:
I think these `if` switches can be removed. They're just duplicating the logic inside the `where` function itself. After you do that, the test will look nicer and the use of parameterize will be justified.

To mark a single parameterize case as expected to fail, you can use `pytest.param` to mark the test as xfail (see [here](https://docs.pytest.org/en/7.1.x/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test)), or be fancier and use `contextlib` to make a dummy `does_not_raise` context, and add a context manager to the test parameterization, like [here](https://docs.pytest.org/en/7.1.x/example/parametrize.html#parametrizing-conditional-raising).

Reply:
To handle the numpy signature (since it's dumb and doesn't let you pass None for x and y), you can just do `np.where(*[x for x in [cond, ift, iff] if x is not None])`. This is valid in all cases without explicit if/else switches.

Reply:
I don't think parametrize makes this test any more readable ;)